zapcore/write_syncer_bench_test.go

```diff
@@ -53,4 +56,51 @@ func BenchmarkMultiWriteSyncer(b *testing.B) {
 			}
 		})
 	})
+	b.Run("4 discarder with buffer", func(b *testing.B) {
```

Comment:
why is this test needed?

Reply:
To compare the performance between two situations below:
1. Writing to the discarder directly
2. Writing to the buffer directly and flushing to the discarder